pkg/controller/registry/grpc/source.go

```diff
@@ -100,10 +105,69 @@ func (s *SourceStore) Get(key registry.CatalogKey) *SourceConn {
 	return &source
 }
 
+func grpcProxyURL(addr string) (*url.URL, error) {
+	// Handle ip addresses
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hardcode fields required for proxy resolution
+	url.Host = addr
+	url.Scheme = "http"
+
+	proxyConfig := httpproxy.FromEnvironment()
+
+	// Override HTTPS_PROXY and HTTP_PROXY with GRPC_PROXY
+	proxyConfig.HTTPProxy = getGRPCProxyEnv()
+	proxyConfig.HTTPSProxy = getGRPCProxyEnv()
```

Comment:
Would it be better to define an `httpproxy.Config` struct here instead? We could optimize cycles with less lines using a struct like so:

```suggestion
	proxyConfig := httpproxy.Config {
		HTTPProxy:  getGRPCProxyEnv()
		HTTPSProxy: getGRPCProxyEnv()
	}
```

Reply:
We need the NO_PROXY variable as well and I thought it was easier to use a defined function than to recreate the code. But I'll go ahead and change it since it'll be easier to read.

Reply:
Addressed.